src/framework/linalg/almost_equal.hpp

```diff
@@ -39,6 +39,26 @@ bool almost_equal(T f1, T f2,
          max_relative_diff * std::max(std::abs(f1), std::abs(f2));
 }
 
+// Complex numbers are probably not considered scalar, so we need another
+// implementation to deal with them
+template <typename complex_t, typename = enable_if_complex_t<complex_t>>
+bool complex_almost_equal(complex_t f1, complex_t f2,
```

Comment:
Complex shouldn't need a special almost equal. The `abs` function handles that case.

Reply:
Unfortunately, the `max_diff` (`std::numeric_limits<std::complex<double>>`) won't work. So as talked, let's create an overloaded function for complex types.